tensorbay/dataset/tests/test_data.py

```diff
@@ -93,26 +89,38 @@ def test_loads_dumps(self, loads) -> None:
 class TestData:
     """Test Data class."""
 
+    def test_init(self) -> None:
+        local_path = "test.json"
+        target_remote_path = "A/test.json"
+        timestamp = 1614667532
+
+        data = Data(local_path, target_remote_path=target_remote_path, timestamp=timestamp)
+        assert data.path == local_path
+        assert data.target_remote_path == target_remote_path
+        assert data.timestamp == timestamp
+
     def test_loads_dumps(self) -> None:
         data = Data.loads(_DATA)
         assert data.dumps() == _DATA
 
-    def test_get_url(self, data_object) -> None:
-        with pytest.raises(ValueError):
-            data_object.get_url()
-
-    def test_get_tbrn(self, data_object) -> None:
-        with pytest.raises(ValueError):
-            data_object.tbrn
+    def test_target_remote_path(self) -> None:
+        target_remote_path = "test2.json"
+        data_object = Data("test.json")
+        data_object.target_remote_path = target_remote_path
+        assert data_object.target_remote_path == target_remote_path
 
-    def test_remote_path(self, data_object) -> None:
-        assert data_object.remote_path == "test.json"
 
-        data_object.remote_path = "test2.json"
-        assert data_object.remote_path == "test2.json"
+class TestRemoteData:
+    """Test RemoteData class."""
 
-    def test_local_path(self, data_object) -> None:
-        assert data_object.local_path == "test.json"
+    def test_init(self) -> None:
+        remote_path = "A/test.json"
+        timestamp = 20210302
+        remote_data = RemoteData(remote_path, timestamp=timestamp)
+        assert remote_data.path == remote_path
+        assert remote_data.timestamp == timestamp
 
-        data_object = Data.loads({"fileuri": "tb:datasetname:segmentname://test.json"})
-        assert data_object.local_path == ""
+    def test_get_url(self) -> None:
```

Comment:
For testing `get_url()`
you can send a lambda as `url_getter` into `RemoteData`